### PR TITLE
Add SHA-2 self-signed cert and use SecureString as password parameter

### PR DESCRIPTION
**Pull request by ciaozhang**
Fix security bugs:
1) Do not use weak crypt algorithm:
     Using SHA256 and create a new self-signed cert
2) Do not use plain password in PKIs
     Using SecureString as password parameter

### CODE REVIEW

**Comment by SameerK-MSFT** [comment]
LGTM. I will leave it to someone in Wilson team to sign off.

**Review by mafurman — APPROVED** [review]

File: test/Microsoft.IdentityModel.Tokens.Tests/RsaCryptoServiceProviderProxyTests.cs [hunk 1]
```diff
@@ -356,10 +356,10 @@ public static TheoryData<RSACryptoServiceProviderProxyTheoryData> RSAVerifyDataT
                 var rsaCsp = new RSACryptoServiceProvider();
                 rsaCsp.ImportParameters(KeyingMaterial.RsaParameters_2048);
 #else
-                var rsaCsp = KeyingMaterial.DefaultCert_2048.PrivateKey as RSACryptoServiceProvider;
+                var rsaCsp = KeyingMaterial.CertSelfSigned1024_SHA256.PrivateKey as RSACryptoServiceProvider;
```
Comment (RojaEnnam) [inline test/Microsoft.IdentityModel.Tokens.Tests/RsaCryptoServiceProviderProxyTests.cs:359]: Is this cert working? If I am not mistaken this is an existing cert that we tried, should it be CertSelfSigned2048_SHA256_2?

Comment (ciaozhang) [reply test/Microsoft.IdentityModel.Tokens.Tests/RsaCryptoServiceProviderProxyTests.cs:359]: This cert works here.

File: build/credscan-exclusion.json [hunk 2]
```diff
@@ -1,61 +1,65 @@
...
-    {
-      "file": "JwtSecurityTokenHandlerTests.cs",
-      "_justification": "File contains tokens that are used only for testing purposes."
-    },
-    {
-      "file": "References.cs",
-      "_justification": "File contains tokens that are used only for testing purposes."
-    }
-  ]
+    "suppressions": [
```
Comment (GeoK) [inline build/credscan-exclusion.json:3]: nit: indentation

File: test/Microsoft.IdentityModel.TestUtils/KeyingMaterial.cs [hunk 3]
```diff
@@ -1057,5 +1071,19 @@ public static JsonWebKey JsonWebKeyX509_2048_As_RSA_With_KeyId
                 return jsonWebKey;
             }
         }
+
+        private static SecureString ConvertToSecureString(string password)
+        {
+            if (password == null)
+                throw new ArgumentNullException("password");
```
Comment (GeoK) [inline test/Microsoft.IdentityModel.TestUtils/KeyingMaterial.cs]: nit: can be replaced with nameof(password)

**Review by GeoK — APPROVED** [review]

**Review by RojaEnnam — APPROVED** [review]

File: test/Microsoft.IdentityModel.Tokens.Tests/RsaCryptoServiceProviderProxyTests.cs (hunk 1 above)
Comment (RojaEnnam) [reply test/Microsoft.IdentityModel.Tokens.Tests/RsaCryptoServiceProviderProxyTests.cs:359]: This test is only checking parameters, doesn't verify. Thanks @ciaozhang for clarifying.

File: build/credscan-exclusion.json (hunk 2 above)
Comment (ciaozhang) [reply build/credscan-exclusion.json:3]: Good catch.

File: test/Microsoft.IdentityModel.TestUtils/KeyingMaterial.cs (hunk 3 above)
Comment (ciaozhang) [reply test/Microsoft.IdentityModel.TestUtils/KeyingMaterial.cs]: Fixed.

**Pull request merged**